compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -3548,11 +3548,18 @@ void OMR::ValuePropagation::transformArrayCloneCall(TR::TreeTop *callTree, TR_Op
    if (!performTransformation(comp(), "%sInlining array clone call [%p] as new array and arraycopy\n", OPT_DETAILS, callNode))
       return;
 
+   TR_OpaqueClassBlock *j9arrayClass = cloneInfo->_clazz;
+   bool isFixedClass = cloneInfo->_isFixed;
+
    TR_OpaqueClassBlock *j9class = comp()->fe()->getComponentClassFromArrayClass(j9arrayClass);
 
    TR::DebugCounter::prependDebugCounter(comp(), TR::DebugCounter::debugCounterName(comp(), "inlineClone.location/array/(%s)", comp()->signature()), callTree);
    int32_t classNameLength;
    char *className = TR::Compiler->cls.classNameChars(comp(), j9arrayClass, classNameLength);
+   if (TR::Compiler->cls.isPrimitiveClass(comp(), j9class))
+      {
+      TR_ASSERT_FATAL(isFixedClass, "Primitive array should be of fixed constraint");
+      }   
```

Comment:
I agree that for arrays of primitives we should generate fixed class constraints, but I'm not sure that a mere bound should be considered _wrong_ per se. We can conclude right here that a primitive element type implies `isFixedClass`, whether or not the original constraint told us so.

Reply:
Removed this assert.